test/query.js

```diff
@@ -990,7 +992,104 @@ describe('then', function() {
     User.filter({}).then(function() {
         done();
     });
+  });
+  it('should return a promise', function(done) {
+    var name = util.s8();
+
+    var Query = thinky.Query;
+    var r = thinky.r;
+    var User = thinky.createModel(modelNames[0], {id: String}, {init: false});
+
+    User.filter({}).then(function() {
+        // no-op
```

Comment:
To make sure that `done` is called from a completed promise chain, rather than just called by `Query` as part of constructing the term.